x-pack/libbeat/management/generate.go

```diff
@@ -140,7 +140,13 @@ func CreateReloadConfigFromInputs(raw []map[string]interface{}) ([]*reload.Confi
 // convinence method for wrapping all the stream transformations needed by the shipper and other inputs
 func createStreamRules(raw *proto.UnitExpectedConfig, streamSource map[string]interface{}, stream *proto.Stream, defaultDataStreamType string, agentInfo *client.AgentInfo, defaultProcessors ...mapstr.M) (map[string]interface{}, error) {
 
-	streamSource = injectIndexStream(defaultDataStreamType, raw, stream, streamSource)
+	// if the index explicitly set it takes precedence
+	if _, exists := streamSource["index"]; exists {
```

Comment:
I'm just wondering how can a user debug this? Maybe adding a log debug informing that `index` is taking precedence.

@rdner do you know how often `createStreamRules` gets called?

If the index/datastream already appears in the event logged by the debug processor or somewhere else, then we don't need any extra logging here.

Reply:
@belimawr we have the rendered Filebeat configuration in the agent diagnostics (`./components/filestream-default/beat-rendered-config.yml`).

Whoever debugs it would see that the `index` is set and there is no `data_stream` on the input level.

You could also see there that we add the data stream information with the `add_fields` processor, for example:

```yaml
        - add_fields:
            fields:
                dataset: filestream.generic
                namespace: default
                type: logs
            target: data_stream
```

This is the way I approached debugging this. I think it's sufficient without adding any logging.

Reply:
Indeed that is enough. I was thinking from a standalone Filebeat, but this feature is limited to running under Elastic Agent, so the rendered config will give enough clues.